README.md

```diff
@@ -175,6 +174,151 @@ $ clarinet check
+
+In the example below, the `asserts!` on line 3 is verifying the `tx-sender`. Because of that check, all inputs are trusted (if the `trusted_sender` option is enabled):
+
+```clarity
+(define-public (take (amount int) (from principal))
+    (let ((balance (- (default-to 0 (get amount (map-get? accounts {holder: from}))) amount)))
+        (asserts! (is-eq tx-sender (var-get bank-owner)) err-unauthorized)
+        (map-set accounts {holder: from} {amount: balance})
+        (stx-transfer? (to-uint amount) (as-contract tx-sender) tx-sender)
+    )
+)
+```
+
+The `callee_filter` option loosens the restriction on passing untrusted data to private functions, and instead, allows checks in a called function to propagate up to the caller. This is helpful, because it allows developers to define input checks in a function that can be reused.
+
+In the example below, the private function `validate` checks its parameter. The public function `save` calls `validate`, and when the `callee_filter` option is enabled, that call to `validate` will count as a check for the untrusted input, `amount`, resulting in no warnings from the check-checker.
+
+```clarity
+(define-public (save (amount uint))
+    (begin
+        (try! (validate amount))
+        (var-set saved amount)
+        (ok amount)
+    )
+)
+
+(define-private (validate (amount uint))
+    (let ((current (var-get saved)))
+        (asserts! (> amount current) err-too-low)
+        (asserts! (<= amount (* current u2)) err-too-high)
+        (ok amount)
+    )
+)
+```
+
+##### Annotations
+
+Sometimes, there is code that the check-checker analysis is unable to determine is safe, but as a developer, you know that it is safe, and want to pass that information to the check-checker to disable warnings that you consider to be false positives. To handle these cases, the check-checker supports several annotations, implemented using "magic comments" in the contract code.
+
+**`#[allow(unchecked_params)]`**
+
+This annotation tells the check-checker that the associated private function is allowed to receive unchecked arguments. It will not generate a warning for calls to this function that pass unchecked inputs. Inside the private function, the parameters are considered unchecked and could generate warnings.
+
+```clarity
+;; #[allow(unchecked_params)]
+(define-private (my-func (amount uint))
+    ...
+)
+```
+
+**`#[allow(unchecked_data)]`**
+
+This annotation tells the check-checker that the following expression is allowed to use unchecked data without warnings. It should be used with care, as it will disable all warnings from the associated expression.
+
+```clarity
+(define-public (dangerous (amount uint))
+    (let ((sender tx-sender))
+        ;; #[allow(unchecked_data)]
+        (as-contract (stx-transfer? amount tx-sender sender))
+    )
+)
+```
+
+**`#[filter(var1, var2)]`**
```

Comment:
What's your take on tagging this annotation as the **recommended** way to deal with warnings?

Reply:
I like that. How about I add this line below the description?

*This is the safest and preferred way to silence warnings that you consider false positives.*

Reply:
👍